calico_containers/tests/st/docker_host.py

```diff
@@ -67,11 +66,19 @@ def execute(self, command, use_powerstrip=False, **kwargs):
         return self._listen(stdin, **kwargs)
 
     def calicoctl(self, command, **kwargs):
+        """
+        Convenience function for abstracting away calling the calicoctl command.
+        """
         calicoctl = "/code/dist/calicoctl %s"
         return self.execute(calicoctl % command, **kwargs)
 
     def start_calico_node(self, ip=None, ip6=None):
+        """
+        Start calico in a container inside a host by calling through to the
+        calicoctl node command.
+        """
         ip = ip or self.ip
+        ip6 = ip6 or self.ip6
```

Comment:
I think we should drop the `ip` and `ip6` arguments and just this method always use the detected IP addresses.

The only place where we're actually passing other IPs is in the arguments parsing test, and I'd argue that test would be easier to understand if it called `host.calicoctl` directly with the arguments under test.

A "would be nice" feature is a way to disable IPv6 support, even if the host container got an IPv6 address, since we do want to test that things don't break if IPv6 isn't available.  That would make it easier to script IPv4 tests that run with and without IPv6 turned on.